src/main/java/gregtech/common/terminal/app/configurator/ConfiguratorApp.java

```diff
@@ -0,0 +1,370 @@
+                    ((CoverCraftingTable) newCover).clearMachineInventory(NonNullList.create());
+                }
+            }
+        }
+        // TODO: Redundant?
+        // Ensure the covers being controlled still exist
+        for (CoverMachineController controlCover : ControlCovers) {
+            if (existingTileEntity.getCoverAtSide(controlCover.getControllerMode().side) == null) {
+                existingTileEntity.removeCover(controlCover.attachedSide);
+            }
+        }
+
+        // Try everything under the sun to get the ***damn block to update visually***
+        existingTileEntity.notifyBlockUpdate();
+        existingTileEntity.markDirty();
+        existingTileEntity.scheduleRenderUpdate();
+
+        // Print logging info
+        if (logging) {
+            if (!log.isEmpty()) {
+                entityPlayer.sendMessage(new TextComponentString("Not all settings were applied:"));
+                entityPlayer.sendMessage(new TextComponentString(String.join("\n", log)));
+            } else {
+                entityPlayer.sendMessage(new TextComponentString("Configuration Applied"));
+            }
+        }
+    }
+
+    private static void copyMetaTileEntity(MetaTileEntity templateTileEntity, MetaTileEntity existingTileEntity) {
+        // Create NBT Tag Compound to paste onto the existing entity
+        NBTTagCompound MTE = new NBTTagCompound();
+        templateTileEntity.clearMachineInventory(NonNullList.create());
+        templateTileEntity.writeToNBT(MTE);
+        // Remove tags that should not be transferred
+        MTE.removeTag("Covers");
+        // All Inventories
+        HashSet<String> keySet = new HashSet<>(MTE.getKeySet());
+        for (String key : keySet) {
+            if (key.toLowerCase().contains("inventory")) {
+                MTE.removeTag(key);
+            }
+        }
+        // TODO: Front facing may not be necessary as it uses the default MTE method
+        // Conflicting Front and Output facings
+        if (!existingTileEntity.isValidFrontFacing(EnumFacing.VALUES[MTE.getInteger("FrontFacing")])) {
+            // If new front facing is not valid, discard it
+            MTE.removeTag("FrontFacing");
+        }
+        // Also, if new output facing conflicts with the existing unchanged front facing, remove it too
+        if (MTE.getInteger("OutputFacing") == existingTileEntity.getFrontFacing().getIndex()) {
+            MTE.removeTag("OutputFacing");
+        }
+        // Special Cases
+        if (templateTileEntity instanceof MetaTileEntityQuantumChest && existingTileEntity instanceof MetaTileEntityQuantumChest) {
+            MTE.removeTag("ItemAmount");
+            MTE.removeTag("ItemStack");
+        }
+        // Finally, transfer the sanitized NBT Compound Tag
+        existingTileEntity.readFromNBT(MTE);
+    }
+
+    private static boolean SubtractFilterIfMissing(List<ItemStack> inventory, CoverBehavior templateCover, CoverBehavior existingCover) {
```

Comment:
method names should be camel case, so `subtractFilterIfMissing` instead of `SubtractFilterIfMissing`, same for the methods after this.